src/Samples/Basic/Temporal.Samples.DesignProcess/Part1_1_TypicalWorkflow.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Hosting;
+
+using Temporal.Common.DataModel;
+using Temporal.Worker.Activities;
+using Temporal.Worker.Hosting;
+using Temporal.Worker.Workflows;
+
+namespace Temporal.Sdk.BasicSamples
+{
+    public class Part1_1_TypicalWorkflow
+    {
+        [Workflow(runMethod: nameof(SayHelloAsync))]
+        public class SayHelloWorkflow
+        {
+            private const string AddresseeNameDefault = "Boss";
+
+            private string _addresseeName = null;
+
+            public async Task SayHelloAsync(WorkflowContext workflowCtx)
+            {
+                var greeting = new SpeechRequest($"Hello, {_addresseeName ?? AddresseeNameDefault}.");                
+                await workflowCtx.Orchestrator.Activities.ExecuteAsync("SpeakAGreeting", greeting);
```

Comment:
Any way you can make this compile-time type safe? Can I just `await workflow.ExecuteActivity<Speak.GreetingAsync>(greeting)` or `await workflow.Activities<Speak>.GreetingAsync(greeting)` or something? Surely there's a dynamic invocation equivalent to https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Proxy.html in .Net where we can have a type-safe wrapper.

Reply:
c# has dynamic objects, but you still need to define an iface to have compile time validation. SUch scaffolding should be supported, but not required.

Reply:
I think this might be a use-case for https://stackoverflow.com/questions/15733900/dynamically-creating-a-proxy-class